datalad/core/local/create.py

```diff
@@ -516,3 +457,115 @@ def __call__(
 
         res.update({'status': 'ok'})
         yield res
+
+
+def _setup_git_repo(path, add_to_git, initopts=None, fake_dates=False):
+    """Create and configure a repository at `path`
+
+    Parameters
+    ----------
+    path: str or Path
+      Path of the repository
+    add_to_git: dict
```

Comment:
`_setup_git_repo` and `_setup_annex_repo` don't care about what information is already in `add_to_git`, so I'd find it clearer if the side channel was avoided by each function dropping the parameter and returning a freshly created dictionary alongside the repo instance.

Reply:
Done!